mantidproject/mantid
# Add toolbar for refl gui instrument view plot

**Description of work.**

This PR adds a toolbar to the embedded instrument viewer display in the ISIS Reflectometry interface and enables toggling between zoom, shape-editing and rectangle-selection modes. As part of this:

- Creation of the ProjectionSurface has been moved to the view because we need to connect to Qt signals from it. The model still provides the instrument actor, samplePos and axis to pass through to create the surface.
- Some work has been done to remove use of QList in the public interface of InstrumentActor and DetXMLFile to make it more compatible with other code that uses std::vector.
- The DrawRegularMode enum has been renamed to EditShapeMode. This mode allows the user to select and edit shapes so this name makes this clearer.

**To test:**

This is enabled in Debug builds only:

- Open the ISIS Reflectometry interface, set instrument to INTER, and select the Preview tab
- Check that the instrument viewer plot has Zoom, Edit and Rectangle-Select buttons above it. They should be greyed out by default.
- Load a run e.g. 13460. The buttons should be enabled with Zoom selected by default. Click and drag in the plot and it should zoom. Right-click and it should un-zoom.
- Try selecting the other buttons and check they work as expected
  - Note that rectangle-select is not fully implemented at the moment; however it should print out the workspace indices of the selected detectors to the log.
- The behaviour should be the same as instrument viewer. In particular, after doing a rectangle-select, it should automatically go into Edit mode to allow you to select/edit the existing selection. You need to re-select rectangle mode in order to add another rectangle

Fixes #32566 

*This does not require release notes* because **the Preview tab is experimental and is currently disabled in the release build**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
